### PR TITLE
Match path if regex starts with /

### PR DESCRIPTION
**Pull request by mishak87**
Pull to #36 issue.
On Windows replace of `/` to `\\\\` is done before matching.

### CODE REVIEW

**Comment by execjosh** [comment]
Wow, thanks!  I won't be able to check this for a few more days; but, thank you very much for helping out :)  Hold on a few more days!

**Comment by execjosh** [comment]
I'd like to get some specs working before I merge this in so we can be sure nothing breaks.

File: lib/scope-name-provider.coffee [hunk 1]
```diff
@@ -44,11 +44,15 @@ class ScopeNameProvider
   # private
   #
 
-  _matchFilename: (filename, opts = {}) ->
+  _matchFilename: (path, opts = {}) ->
+    regexpOptions = if opts.caseSensitive then 'i' else ''
```
Comment (execjosh) [inline lib/scope-name-provider.coffee]: Why is this necessary?

**Comment by execjosh** [comment]
Oh, wait...  There already are some specs for `ScopeNameProvider`.  brain fart

**Comment by execjosh** [comment]
Okay, I've read through your code.  I'm not sure I like requiring that path matchers begin with slashes.  I'll think about this.

**Comment by execjosh** [comment]
Oh, and please add tests to cover the new use cases

**Comment by mishak87** [comment]
I think using Regex in config (not in implementation) for path matching is overkill. Using just [bash glob](http://mywiki.wooledge.org/BashGuide/Patterns) or [go filepath matching](https://golang.org/pkg/path/filepath/#Match) syntax would be more ideal since we are dealing with files not just random strings.

I have added slash to differentiate between path and just filename matching. To avoid possible false positives with configs written for old versions ie.:
```cson
'r.*b.ext': 'source.ruby' # user used path unaware config
# would match: r/b.ext
# not just:    rib.ext
```

Will add tests ASAP (end of this week at most).

**Comment by execjosh** [comment]
I'm open to the idea of bash globbing.  As for v0.5.x, though, RegExp will stay.

BTW, I just implemented the longest-match and tie-breaker logic, as well as deprecating extension matchers.  Those changes conflict with this PR.  Sorry for totally breaking your work effort...

**Comment by execjosh** [comment]
I've opened up #49 to talk about bash globbing :thumbsup:

**Comment by execjosh** [comment]
I'm going to close this for now in favor of #51.
